python/gui/MaaS/templates/maas/dataset_management.html

```diff
@@ -0,0 +1,797 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>OWP MaaS</title>
+    {% load static %}
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js">
+    </script>
+    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js">
+    </script>
+    <script src="{% static 'maas/js/components/confirmDialog.js' %}">
+    </script>
+    <script src="{% static 'maas/js/components/createDatasetForm.js' %}">
+    </script>
+    <script src="{% static 'maas/js/components/datasetOverview.js' %}">
+    </script>
+    <link rel="stylesheet" type="text/css" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/cupertino/jquery-ui.css">
+    <link rel="icon" type="image/gif" href="{% static 'maas/img/favicon.gif' %}">
+    <script>
+        function activate_model(model) {
+            $(".parameters").hide();
+            $("#" + model + "_parameters").show();
+        }
+
+        function buildDatasetsOverviewTable() {
+            $.ajax({
+                url: "/dataset-api",
+                type: 'get',
+                data: {"request_type": "datasets"},
+                success: function(response) {
+                    let tableComp = new DatasetOverviewTable('dataset-manage', 'datasets-table', displayDatasetDetails,
+                        displayDatasetFiles, clickDownloadDataset, clickUploadDataset, clickDeleteDataset);
+                    tableComp.buildTable(response);
+                },
+                error: function(response) {
+                    alert('Received an error');
+                    console.log(response);
+                }
+            });
+        }
+
+        function initDatasetCreateForm() {
+            let category_select = document.getElementById("dataset-create-form-input-category");
+            category_select.selectedIndex = -1;
+            let format_select = document.getElementById("dataset-create-form-input-format");
+            format_select.selectedIndex = -1;
+        }
+
+        function initGuiViews() {
+            buildDatasetsOverviewTable();
+            initDatasetCreateForm();
+        }
+
+        function changeCreateDatasetFormatSelection(selection) {
+            let dy_div = document.getElementById("dataset-create-form-div-dynamic-inputs");
+            for (let i = 0; i < dy_div.children.length; i++) {
```

Comment:
I think `dy_div.children[i].remove()` changes the length of `dy_div.children`. Instead, I think we want the following:

```javascript
// remove all dynamic input child elements
while (dy_div.firstChild) {
  dy_div.removeChild(dy_div.firstChild);
}
```